### PR TITLE
HBASE-27036 Displays the number of decommissioned region server for status command

### PR DESCRIPTION
**Pull request by tomscut**
JIRA: HBASE-27036.

Displays the number of decommissioned region server for status command.
image

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 40s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 40s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 39s |  branch has no errors when running spotless:check.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 58s |  the patch passed  |
| -0 :warning: |  rubocop  |   0m  5s |  The patch generated 1 new + 402 unchanged - 1 fixed = 403 total (was 403)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  spotless  |   0m 37s |  patch has no errors when running spotless:check.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  8s |  The patch does not generate ASF License warnings.  |
|  |   |   9m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4431 |
| Optional Tests | dupname asflicense javac spotless rubocop |
| uname | Linux 64d07c7023ec 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 369adcdf35 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| rubocop | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 54 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/console |
| versions | git=2.17.1 maven=3.6.3 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 13s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 13s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 37s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 11s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   6m 47s |  hbase-shell in the patch passed.  |
|  |   |  14m 58s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4431 |
| Optional Tests | javac javadoc unit |
| uname | Linux 4edf028134b1 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 369adcdf35 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/testReport/ |
| Max. process+thread count | 1543 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 36s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 16s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 55s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m 41s |  hbase-shell in the patch passed.  |
|  |   |  16m 43s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4431 |
| Optional Tests | javac javadoc unit |
| uname | Linux 9d51e5487c7b 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 369adcdf35 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/testReport/ |
| Max. process+thread count | 1390 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by tomscut** [comment]
Hi @apurtell @virajjasani @pankaj72981 , please take a look. Thanks.

File: hbase-shell/src/main/ruby/hbase/admin.rb [hunk 1]
```diff
@@ -1043,6 +1043,7 @@ def status(format, type)
       else
         puts "1 active master, #{cluster_metrics.getBackupMasterNames.size} backup masters,
               #{cluster_metrics.getLiveServerMetrics.size} servers,
+              #{@admin.listDecommissionedRegionServers.size} decommissioned,
```
Comment (Apache9) [inline hbase-shell/src/main/ruby/hbase/admin.rb:1046]: We do not have this information in cluster metrics?

Comment (tomscut) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1046]: Thanks @Apache9 for your review.

 Now, we do not have `decommissionedRegionServers` in cluster metrics. But we have metrics `numDrainingRegionServers`  and `drainingRegionServers`. 

There is a method `listDecommissionedRegionServers` in `Admin`,  so called directly here.

Comment (Apache9) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1046]: Then let's add it to the cluster metrics?

Comment (tomscut) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1046]: Thanks for your comment. 

At first, I wanted to add `drainingRegionServers` to cluster metrics, but I found that `drainingRegionServers` already exist. In most cases, `drainingRegionServers` and `decommissionedRegionServers` may represent the same list of regionservers, so I didn't add this metric.

Comment (pankaj72981) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1046]: Cluster metrics approach is better than calling additional admin API (includes a separate RPC). 

Decommissioned and draining RegionServers are same,
https://github.com/apache/hbase/blob/1be2c83b3dc8306050f67828c9c2e361f19481c6/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java#L3821

Add decommissioned/draining RegionServer to ClusterMetrics class, currently it is not there.

Comment (tomscut) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1046]: > Cluster metrics approach is better than calling additional admin API (includes a separate RPC).
> 
> Decommissioned and draining RegionServers are same,
> 
> https://github.com/apache/hbase/blob/1be2c83b3dc8306050f67828c9c2e361f19481c6/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java#L3821
> 
> Add decommissioned/draining RegionServer to ClusterMetrics class, currently it is not there.

Thanks @pankaj72981 for your comments. I found `drainingRegionServers` already exist, we can see HBASE-25460. Do you mean to change `drainingRegionServers` to `decommissionedRegionServers `?

Comment (pankaj72981) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1046]: HBASE-25460 added draininigRegionServers as a JMX metric. Can you access it via ClusterMetrics class?

Comment (tomscut) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1046]: > [HBASE-25460](https://issues.apache.org/jira/browse/HBASE-25460) added draininigRegionServers as a JMX metric. Can you access it via ClusterMetrics class?

Thanks @pankaj72981 for your comment. 

I will try to add this to ClusterMetrics. Do you think `draininigRegionServers` or `decommissionedRegionServers` more appropriate? @Apache9 Please also take a look. Thank you.

**Comment by tomscut** [comment]
Hi @Apache9 @pankaj72981 , I added `decommissionedRegionServers` to ClusterMetrics. I deployed a cluster with master branch, and it works. PTAL. Thanks a lot.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 24s |  master passed  |
| +1 :green_heart: |  compile  |   4m  8s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  2s |  master passed  |
| -1 :x: |  spotless  |   0m 37s |  branch has 22 errors when running spotless:check, run spotless:apply to fix.  |
| +1 :green_heart: |  spotbugs  |   4m 45s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m  9s |  the patch passed  |
| +1 :green_heart: |  cc  |   4m  9s |  the patch passed  |
| -0 :warning: |  javac  |   0m 38s |  hbase-client generated 1 new + 122 unchanged - 1 fixed = 123 total (was 123)  |
| +1 :green_heart: |  checkstyle  |   1m  2s |  the patch passed  |
| -0 :warning: |  rubocop  |   0m  4s |  The patch generated 1 new + 402 unchanged - 1 fixed = 403 total (was 403)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 47s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  hbaseprotoc  |   1m 38s |  the patch passed  |
| -1 :x: |  spotless  |   0m 36s |  patch has 22 errors when running spotless:check, run spotless:apply to fix.  |
| +1 :green_heart: |  spotbugs  |   4m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   |  46m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4431 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool rubocop |
| uname | Linux 0e730345fbb3 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 194f8c2c41 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-general-check/output/branch-spotless.txt |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-general-check/output/diff-compile-javac-hbase-client.txt |
| rubocop | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 64 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 13s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 35s |  master passed  |
| +1 :green_heart: |  compile  |   2m  1s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 42s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  6s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 34s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  1s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 41s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 39s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 27s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 198m 34s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   6m 53s |  hbase-shell in the patch passed.  |
|  |   | 229m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4431 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux ed5262202b13 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 194f8c2c41 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/testReport/ |
| Max. process+thread count | 2785 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 42s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 49s |  master passed  |
| +1 :green_heart: |  compile  |   1m 58s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 43s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 46s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 17s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 17s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 45s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  6s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 39s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 10s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 243m  3s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  unit  |   8m  1s |  hbase-shell in the patch passed.  |
|  |   | 278m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4431 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 205a266029c9 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 194f8c2c41 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/testReport/ |
| Max. process+thread count | 2501 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by tomscut** [comment]
Thanks @Apache9 for the review.

**Review by pankaj72981 — APPROVED** [review]

**Comment by tomscut** [comment]
Thanks @pankaj72981 for your review.

**Comment by tomscut** [comment]
Hi @virajjasani , please also take a look. Thank you.

**Comment by tomscut** [comment]
Hi @pankaj72981 , could you please merge this? Thanks.

**Comment by Apache9** [comment]
Let me trigger another pre commit run.

**Review by virajjasani — APPROVED** [review]

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 10s |  master passed  |
| +1 :green_heart: |  compile  |   4m  2s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  1s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 41s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   4m 30s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 12s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m  9s |  the patch passed  |
| +1 :green_heart: |  cc  |   4m  9s |  the patch passed  |
| -0 :warning: |  javac  |   0m 39s |  hbase-client generated 1 new + 122 unchanged - 1 fixed = 123 total (was 123)  |
| +1 :green_heart: |  checkstyle  |   1m  0s |  the patch passed  |
| -0 :warning: |  rubocop  |   0m  4s |  The patch generated 1 new + 402 unchanged - 1 fixed = 403 total (was 403)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 24s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  hbaseprotoc  |   1m 37s |  the patch passed  |
| +1 :green_heart: |  spotless  |   0m 41s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   4m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   |  45m 50s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4431 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool rubocop |
| uname | Linux a855ee6c9ca6 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5c116d239f |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| javac | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/3/artifact/yetus-general-check/output/diff-compile-javac-hbase-client.txt |
| rubocop | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/3/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 64 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4431/3/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**